contracts/DCAFeeManager/DCAFeeManager.sol

```diff
@@ -12,7 +12,50 @@ contract DCAFeeManager is Governable, IDCAFeeManager {
   /// @inheritdoc IDCAFeeManager
   IDCAHub public immutable hub;
 
-  constructor(IDCAHub _hub, address _governor) Governable(_governor) {
+  TargetTokenShare[] internal _distribution;
+
+  constructor(
+    IDCAHub _hub,
+    TargetTokenShare[] memory _distributionToSet,
+    address _governor
+  ) Governable(_governor) {
     hub = _hub;
+    setTargetTokensDistribution(_distributionToSet);
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function targetTokensDistribution() external view returns (TargetTokenShare[] memory) {
+    return _distribution;
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function setTargetTokensDistribution(TargetTokenShare[] memory _newDistribution) public {
+    // TODO: Make sure only specific users can set this distribution
+
+    uint256 _currentTargetTokens = _distribution.length;
+    uint256 _min = _currentTargetTokens < _newDistribution.length ? _currentTargetTokens : _newDistribution.length;
+
+    uint16 _assignedShares;
+    for (uint256 i; i < _min; i++) {
+      // Rewrite storage
+      _assignedShares += _newDistribution[i].shares;
+      _distribution[i] = _newDistribution[i];
+    }
+
+    if (_currentTargetTokens < _newDistribution.length) {
+      // If have more tokens than before, then push
+      for (uint256 i = _min; i < _newDistribution.length; i++) {
+        _assignedShares += _newDistribution[i].shares;
+        _distribution.push(_newDistribution[i]);
+      }
+    } else if (_currentTargetTokens > _newDistribution.length) {
+      // If have less tokens than before, then remove extra tokens
+      for (uint256 i = _min; i < _currentTargetTokens; i++) {
+        _distribution.pop();
+      }
+    }
```

Comment:
I think you can avoid re-declaring `i` here. If you declare `i` out of the first `for` scope, then at the end of the loop `i = min`, right? and you can avoid the re-declaration.

Reply:
Great catch!